src/components/TodoList/TodoList.tsx

```diff
@@ -1,44 +1,125 @@
+import classNames from 'classnames';
 import React from 'react';
+import { Requirement } from '../../enums/Requirement';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
 import './TodoList.scss';
 
-export const TodoList: React.FC = () => (
-  <div className="TodoList">
-    <h2>Todos:</h2>
-
-    <div className="TodoList__list-container">
-      <ul className="TodoList__list">
-        <li className="TodoList__item TodoList__item--unchecked">
-          <label>
-            <input type="checkbox" readOnly />
-            <p>delectus aut autem</p>
-          </label>
-
-          <button
-            className="
-              TodoList__user-button
-              TodoList__user-button--selected
-              button
-            "
-            type="button"
-          >
-            User&nbsp;#1
-          </button>
-        </li>
-
-        <li className="TodoList__item TodoList__item--checked">
-          <label>
-            <input type="checkbox" checked readOnly />
-            <p>distinctio vitae autem nihil ut molestias quo</p>
-          </label>
-
-          <button
-            className="TodoList__user-button button"
-            type="button"
+type Props = {
+  todos: Todo[];
+  selectedUserId: User['id'];
+  onSelect: (newSelectedUserId: User['id']) => void;
+};
+
+type State = {
+  titleQuery: string;
+  todoRequirement: Requirement;
+};
+
+export class TodoList extends React.Component<Props, State> {
+  state: State = {
+    titleQuery: '',
+    todoRequirement: Requirement.ALL,
+  };
+
+  handleChange = (
+    event: React.ChangeEvent<HTMLInputElement> | React.ChangeEvent<HTMLSelectElement>,
+  ) => {
+    const { name, value } = event.target;
+
+    this.setState({
+      [name]: value,
+    } as Pick<State, 'titleQuery'>);
+  };
+
+  getVisibleTodos = () => {
+    const { todos } = this.props;
+    const { titleQuery } = this.state;
+
+    const meetRequirement = (todo: Todo) => {
```

Comment:
```suggestion
    const hasCorrectStatus = (todo: Todo) => {
```
No strong opinion though

Reply:
I agree with you and trust you more than with/*blank* anyone else, so okay, I'll rename it)

Reply:
All the mentors and couch are here to help you, so you should trust all of us). Although it's possible that each of us can make mistakes for sure.